### PR TITLE
Sticky Custom Post Types plugin Refactor

### PR DESCRIPTION
**Pull request by renatonascalves**
This plugin was very useful for a client. This PR is to update it to:

- account for code best practices;
- add the stick class on sticky posts;
- add internationalization;
- Other minor fixes and updates.

### CODE REVIEW

File: sticky-custom-post-types/sticky-custom-post-types.php [hunk 1]
```diff
@@ -25,84 +36,260 @@
...
+		 * @return void
+		 */
+		public function set_sticky_post_types() {
+			$post_types = get_post_types( [
+				'_builtin' => false,
+				'public'   => true,
+			], 'names' );
+
+			if ( empty( $post_types ) ) {
+				echo '<p>' . __( 'No public custom post types found.', 'sticky-custom-post-types' ) . '</p>'; // WPCS: XSS ok.
```
Comment (rinatkhaziev) [inline sticky-custom-post-types/sticky-custom-post-types.php]: Could you please update __ to esc_html__?

File: sticky-custom-post-types/sticky-custom-post-types.php [hunk 2]
```diff
@@ -25,84 +36,260 @@
...
+			// Bail if user cannot edit posts.
+			if ( ! current_user_can( 'edit_others_posts' ) ) {
+				return;
+			}
 
-		$super_sticky_post_types = super_sticky_post_types();
+			// Add the meta box.
+			add_meta_box(
+				'super_sticky_meta',
+				__( 'Sticky', 'ione3' ),
```
Comment (rinatkhaziev) [inline sticky-custom-post-types/sticky-custom-post-types.php]: Looks like `ione3` is a copy-paste that needs to be fixed.

File: sticky-custom-post-types/sticky-custom-post-types.php [hunk 3]
```diff
@@ -25,84 +36,260 @@
...
-		if(current_user_can('edit_others_posts'))
-			add_meta_box('super_sticky_meta', 'Sticky', 'super_sticky_meta', $post_type, 'side', 'high');
-}
+		/**
+		 * Sticky Description.
+		 *
+		 * @return void
+		 */
+		public function sticky_description() {
+			echo '<p>' . __( 'Enable support for sticky custom post types.', 'sticky-custom-post-types' ) . '</p>'; // WPCS: XSS ok.
```
Comment (rinatkhaziev) [inline sticky-custom-post-types/sticky-custom-post-types.php]: Could you please update __ to esc_html__?

**Review by rinatkhaziev — CHANGES_REQUESTED** [review]
Hi, we reviewed the code, there are some minor escaping issues that need to be fixed.

File: sticky-custom-post-types/sticky-custom-post-types.php (hunk 1 above)
Comment (renatonascalves) [reply sticky-custom-post-types/sticky-custom-post-types.php]: Updated!

File: sticky-custom-post-types/sticky-custom-post-types.php (hunk 2 above)
Comment (renatonascalves) [reply sticky-custom-post-types/sticky-custom-post-types.php]: Sorry! Updated!

File: sticky-custom-post-types/sticky-custom-post-types.php (hunk 3 above)
Comment (renatonascalves) [reply sticky-custom-post-types/sticky-custom-post-types.php]: Updated!

File: sticky-custom-post-types/sticky-custom-post-types.php [hunk 4]
```diff
@@ -25,84 +36,263 @@
...
+				'public'   => true,
+			], 'names' );
 
-add_action('admin_init', 'super_sticky_add_meta_box');
+			if ( empty( $post_types ) ) {
+				echo '<p>' . esc_html__( 'No public custom post types found.', 'sticky-custom-post-types' ) . '</p>'; // WPCS: XSS ok.
+			}
+
+			// Filter the post types supported.
+			$post_types = apply_filters( 'sticky_supported_custom_post_types', $post_types );
```
Comment (renatonascalves) [inline sticky-custom-post-types/sticky-custom-post-types.php:136]: This adds a way for developers to remove, from selection, certain **valid** custom post types programatically.

**Comment by renatonascalves** [comment]
@rinatkhaziev Could you take another look? =)

**Review by rinatkhaziev — APPROVED** [review]

**Comment by renatonascalves** [comment]
@rinatkhaziev it seems it was approved for merge [here](https://wordpressvip.zendesk.com/hc/en-us/requests/85505).

=)

**Comment by renatonascalves** [comment]
For posterity, this PR implementation was added but with a different slug: https://github.com/svn2github/wordpress-vip-plugins/tree/master/sticky-custom-post-types-1.3

Thanks! =)

**Pull request closed without merging**